### ISSUE TITLE
Cannot compile Lazarus 2.0.8 with FPC 3.2.0

### ISSUE THREAD

**Issue by gcarreno** [LongDirtyAnimAlf/fpcupdeluxe#259]
Hey Don,

From #257 you know I've started with a clean `svn checkout` of `FPC` 3.2.0 and I tried to compile `Lazarus` 2.0.8 from that and it's not compiling due to `chm`:

```
{...}
(3104) Compiling unusedunitsdlg.pas
(3104) Compiling idefpcinfo.pas
(3104) Compiling ideinfodlg.pas
(3104) Compiling ideinfoneedbuild.pas
(3104) Compiling initialsetupdlgs.pas
(3104) Compiling makeresstrdlg.pas
(3104) Compiling codeexplorer.pas
(3104) Compiling buildfiledlg.pas
(3104) Compiling procedurelist.pas
(3104) Compiling extractprocdlg.pas
(3104) Compiling findrenameidentifier.pas
(3104) Compiling emptymethodsdlg.pas
(3104) Compiling useunitdlg.pas
(3104) Compiling findoverloadsdlg.pas
(3104) Compiling editorfilemanager.pas
(3104) Compiling cleandirdlg.pas
(3104) Compiling showdeletingfilesdlg.pas
(3104) Compiling compatibilityrestrictions.pas
(3104) Compiling restrictionbrowser.pas
(3104) Compiling projectwizarddlg.pas
(3104) Compiling sourcefilemanager.pas
(3104) Compiling notifyprocessend.pas
(10001) PPU Loading /home/gcarreno/FreePascal_stable/fpc/units/x86_64-linux/chm/chmreader.ppu
(10011) PPU Source: chmreader.pas not available
(10028) Recompiling chmreader, checksum changed for /home/gcarreno/FreePascal_stable/lazarus/components/sparta/generics/lib/x86_64-linux/generics.collections.ppu
make[2]: *** [Makefile:4574: idepkg] Error 2
make[1]: *** [Makefile:3385: idepkg] Error 2
/home/gcarreno/FreePascal_stable/fpc/units/x86_64-linux/chm/chmreader.ppu:chmreader.pas(80,12) Fatal: (10022) Can't find unit chmreader used by ChmLangRef
Fatal: (1018) Compilation aborted
Error: /home/gcarreno/FreePascal_stable/fpc/bin/x86_64-linux/ppcx64 returned an error exitcode
make: *** [Makefile:3390: useride] Error 2
Error: (lazarus) Build IDE: stopped with exit code 2
Error: (lazarus) Building IDE: Building IDE failed.
fpcupdeluxe: ERROR: LazarusNativeInstaller (BuildModuleCustom: UserIDE): make returned exit status #512


ERROR: Fpcupdeluxe fatal error !
Sequencer (UserIDE): Failure running fpcupdeluxe: error executing sequence UserIDE
Sequencer (Lazarus): Failure running fpcupdeluxe: error executing sequence Lazarus
Sequencer (Only): Failure running fpcupdeluxe: error executing sequence Only
```
So I decided to delete the `lazarus` folder to see if I had junk from previous compiles that would mess it up. After a complete `svn checkout` from nothing, same issue.

I've had this issue on trunk, intermittently, so I've never bother to get to the root of it since trunk is not stable and it's changing all the time.
But having this issue on both `FPC` and `Lazarus` stable, means that I'll only solve this if `Lazarus` ups the stable from 2.0.8 to something else.

Is this a known issue with an easy fix?

Cheers,
Gus

**Comment by gcarreno** [comment]
Hey Don,

I was able to compile `Lazarus` if I rename the `config_lazarus` to something else, so the basic `Lazarus` without the Docking.

As soon I add the `anchordocking` component, the `chm` issue pops up.

Hope this helps in any way.

Cheers,
Gus

**Comment by gcarreno** [comment]
Hey Don,

After reading that error text I put on this issue and the realization that adding Anchor Docking makes it go BOOM, it as, definitely, something to do with the `sparta` stuff.

I was able to get my `Lazarus` installation up to date with all my components and it's fine. I can't add the docking however or it doesn't work.

What really baffles me is what does the `sparta` generics have to do with the `chmreader`?!?!?!?!?
Do those `sparta` generics mess with the `FPC` generics?!?!?!

As you can see I'm completely lost.

Cheers,
Gus

**Comment by LongDirtyAnimAlf** [comment]
This issue has already been reported on the bug-tracker AFAIR. But I cannot find it.
As this problem is not related to fpcupdeluxe, I am closing.
If I know more about this issue, I will report back here.

**Comment by gcarreno** [comment]
Hey Don,

I've also chased this rabbit a bit further and in the bug tracker it's a closed issue with Rev around `61xxx`.
Which is a tad strange cuz `Lazarus 2.0.10` is at  Rev `63526` and it still goes boom with `sparta_generics` and `chm` but `Lazarus trunk` at Rev `63540` is absolutely fine.

One of the suggestions, on the forum and on the bug tracker, is to remove `sparta_generics` from `sparta_tools.lpk` and `sparta_mdi.lpk`, which in my opinion does not address the main issue. I've downloaded a patch file from the forum or the bug tracker, i forget, that does this.

As a curiosity, `Lazarus 2.0.10` is still going BOOM with `sparta` and `chm`.

Will persist a bit more on this just to ease my mind.

Cheers,
Gus
